src/compiler/program.ts

```diff
@@ -1675,7 +1675,7 @@ namespace ts {
             });
 
             if (packageId) {
-                const packageIdKey = `${packageId.name}@${packageId.version}`;
+                const packageIdKey = `${packageId.name}/${packageId.subModuleName}@${packageId.version}`;

```

Comment:
Just to confirm, we're fine with producing an ugly string when `packageId.subModuleName` is empty?  It appears to only be used as a dictionary key, so I'm assuming we are.